packages/worker/src/asg.ts

```diff
@@ -31,3 +31,32 @@ export const updateScaleInProtection = async (
 		throw error;
 	}
 };
+
+export const getInstanceLifecycleState = async (
+	autoScalingClient: AutoScalingClient,
+	stage: string,
+	instanceId: string,
+) => {
+	try {
+		if (stage !== 'DEV') {
+			const input = {
+				InstanceIds: [instanceId],
+			};
+			const command = new DescribeAutoScalingInstancesCommand(input);
+			const result = await autoScalingClient.send(command);
+			const lifecycleState = result.AutoScalingInstances?.find(
+				(i) => i.InstanceId === instanceId,
+			)?.LifecycleState;
+			if (lifecycleState === undefined)
+				throw new Error('Could not find instance lifecycle state!');
```

Comment:
maybe include curly braces https://softwareengineering.stackexchange.com/questions/16528/single-statement-if-block-braces-or-no

Reply:
agreed - this felt wrong